### DISCUSSION TITLE
Deprecations in semconv

### DISCUSSION THREAD

**Discussion by IdlePhysicist** [open-telemetry/opentelemetry-go#5853] [Q&A] [answered]
Howdy, firstly if this forum is not the correct place this discussion please direct me to the more appropriate venue.

I was recently patching an application that was using `go.opentelemetry.io/otel/semconv/v1.24.0` to use `go.opentelemetry.io/otel/semconv/v1.26.0` instead.  There were several deprecated constants removed in the 1.26 release that I was able to handle by updating to the recommended replacement e.g. `semconv.HTTPURLKey` -> `semconv.URLFullKey`.

But one eluded me which is `semconv. HTTPRequestContentLengthKey` which was removed in 1.26.  The docstring suggests that I use `http.request.header.content-length` instead, but I am unable to locate a constant named `HTTPRequestHeaderContentLengthKey`  or any constant / function starting with `HTTPRequest`.

In the end I wrote my own but am I going crazy and just not able to see what is in front of me?

https://pkg.go.dev/go.opentelemetry.io/otel@v1.30.0/semconv/v1.24.0#HTTPRequestContentLengthKey

**Comment by pellared** [comment]
First of all, you are not crazy 😉 

The semconv packages are generated from https://github.com/open-telemetry/semantic-conventions.
It looks like there is a bug there.  The `http.response.header.content-length` attribute is missing. See: https://github.com/search?q=repo%3Aopen-telemetry%2Fsemantic-conventions%20http.response.header.content-length&type=code

I am creating an issue out of this discussion.

**Reply by pellared** [reply, marked as answer]
I created https://github.com/open-telemetry/semantic-conventions/issues/1439

**Reply by IdlePhysicist** [reply]
Always good to know that I'm not crazy (yet).
